exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java

```diff
@@ -128,55 +148,184 @@ void createTestKitWithoutServices() {
     assertThrows(exceptionType, testKitBuilder::build);
   }
 
-  public static final class TestServiceModule extends AbstractServiceModule {
+  @Test
+  void initializationChangesState() {
+    TestKit testKit = TestKit.forService(TestServiceModule.class);
+    Map<HashCode, String> map = testKit.withSnapshot((view) -> {
+      TestService testService = testKit.getService(TestService.SERVICE_ID, TestService.class);
+      TestSchema testSchema = testService.createDataSchema(view);
+      ProofMapIndexProxy<HashCode, String> proofMapIndexProxy = testSchema.testMap();
+      return toMap(proofMapIndexProxy);
+    });
+    assertThat(map).hasSize(1);
+    String initialValue = map.get(TestService.INITIAL_ENTRY_KEY);
+    assertThat(initialValue).isEqualTo(TestService.INITIAL_ENTRY_VALUE);
+  }
 
-    private static final TransactionConverter THROWING_TX_CONVERTER = (tx) -> {
-      throw new IllegalStateException("No transactions in this service: " + tx);
-    };
+  @Test
+  void createEmptyBlock() {
+    TestKit testKit = TestKit.forService(TestServiceModule.class);
+    Block block = testKit.createBlock();
+    assertThat(block.getNumTransactions()).isEqualTo(0);
 
-    @Override
-    protected void configure() {
-      bind(Service.class).to(TestService.class).in(Singleton.class);
-      bind(TransactionConverter.class).toInstance(THROWING_TX_CONVERTER);
-    }
+    testKit.withSnapshot((view) -> {
+      Blockchain blockchain = Blockchain.newInstance(view);
+      assertThat(blockchain.getHeight()).isEqualTo(1);
+      assertThat(block).isEqualTo(blockchain.getBlock(1));
+      return null;
+    });
   }
 
-  public static final class TestServiceModule2 extends AbstractServiceModule {
+  @Test
+  void afterCommitSubmitsTransaction() {
+    TestKit testKit = TestKit.forService(TestServiceModule.class);
 
-    private static final TransactionConverter THROWING_TX_CONVERTER = (tx) -> {
-      throw new IllegalStateException("No transactions in this service: " + tx);
-    };
+    Block block = testKit.createBlock();
```

Comment:
It is unclear why this is invoked.

Reply:
We want to create a block so that `afterCommit` is invoked and we can test it (by verifying that it indeed submits a transaction). Added a comment